src/resources/ble_esp32.cc

```diff
@@ -2133,25 +2140,30 @@ PRIMITIVE(deploy_service) {
     characteristic_index++;
   }
 
-  struct ble_gatt_svc_def gatt_svcs[2] = {
-    {
-      .type = BLE_GATT_SVC_TYPE_PRIMARY,
-      .uuid = service_resource->ptr_uuid(),
-      .includes = 0,
-      .characteristics = gatt_svr_chars
-    },
+
+  auto gatt_svcs = static_cast<ble_gatt_svc_def*>(calloc(2, sizeof(ble_gatt_svc_def)));
```

Comment:
We are using the fact that `BLE_GATT_SVC_TYPE_END` equals 0. It's probably never going to change, but I think it helps readability to assert it.
```suggestion
  auto gatt_svcs = static_cast<ble_gatt_svc_def*>(calloc(2, sizeof(ble_gatt_svc_def)));
  static_assert(BLE_GATT_SVC_TYPE_END == 0, "Unexpected TYPE_END value");
```

Alternatively, or in addition, we could set the last entry:
```
gatt_svcs[1].type = BLE_GATT_SVC_TYPE_END;
```

Reply:
Done.